% !TEX TS-program = pdflatex
% !BIB TS-program = biber
\documentclass[12pt,letterpaper]{article}

% Layout and formatting
\usepackage[margin=1in]{geometry}
\usepackage{acronym}
\usepackage{csquotes}
\usepackage{fixltx2e}
\usepackage{url}
\usepackage{mathptmx}
\usepackage{setspace}
\usepackage{units}
\frenchspacing

% Graphics
\usepackage{graphicx}
\usepackage[pdf]{pstricks}
\psset{unit=1in,linewidth=0.02,arrows=C-C,arrowsize=0.2}

% Bibliography
\usepackage[american]{babel}
\usepackage[backend=biber,style=apa]{biblatex}
\DeclareLanguageMapping{american}{american-apa}
\bibliography{bibliography.bib}
\newcommand{\citeapos}[1]{\citeauthor{#1}'s (\citeyear{#1})}
\usepackage{doi}

% Figures, tables, and captions
\usepackage{float}
\usepackage{booktabs}
\usepackage{caption}
\captionsetup{labelfont=bf,font=small,labelsep=period}
\usepackage{subcaption}
\usepackage[rightcaption]{sidecap}

% Math
\usepackage{amsmath}
\usepackage{algorithm2e}
\usepackage{mathtools}
\usepackage{commath}

\title{Bayesian Reconstruction of Coevolutionary Histories}

% Acronyms
\acrodef{GTR}{general time reversible}
\acrodef{MCMC}{Markov chain Monte Carlo}


\begin{document}

\begin{titlepage}
\null
\vfil
\let\newpage\relax
\maketitle
\vfil
\centering
\begin{pspicture}(18,12)
\psset{unit=0.5cm,linewidth=0.2}
\psline[linecolor=blue](0,0)(10,10)
\psline[linecolor=blue](4,0)(2,2)
\psline[linecolor=blue](8,0)(4,4)
\psline[linecolor=blue](12,0)(14,2)
\psline[linecolor=blue](16,0)(8,8)
\psline[linecolor=red](1,0)(11,10)
\psline[linecolor=red,arrows=-o](17,0)(9,8)
\psline[linecolor=red,arrows=-o](13,0)(15,2)
\psline[linecolor=red](7,3)(14,3)
\psline[linecolor=red,arrows=<-](10,3)(14,3)
\psline[linecolor=red](10,0)(7,3)
\psline[linecolor=red,arrows=-o](9,0)(5,4)
\psline[linecolor=red,arrows=-o](4,1)(3,2)
\rput{135}(4,1){\LARGE\textcolor{red}{\textsf{\textbf{x}}}}
\psline[linecolor=red](18,0)(17,1)
\psline[linecolor=red,arrows=*-](16,1)(17,1)
\end{pspicture}
\vfil
\end{titlepage}

\newpage

\doublespacing

\section*{Introduction}

%Considerable . To date, there are a number of tools available for .
%
%It is important to note that the host-symbiont [reconciliation] problem is largely similar to the gene tree--species tree [reconciliation] problem. Accordingly, it is worth comparing these 
%
%\textcite{Akerborg:2009} compare [reconciliation] methods to the evolution of methods for phylogenetic evidence, highlighting the shift to the probabilistic, and in particular, Bayesian approach.
%
%However, problems similar to those of parsimony inference of phylogeny also plague parsimony inference of cophylogenetic mappings.
%
%A likelihood implementation has many advantages \parencite{Charleston:2009}.

%Symbiotic relationships, or interactions between organisms of different species, occur in all domains of life
%
%The emergence of DNA sequencing techniques in the late twentieth century encouraged the development of statistical methods to compare sequences .
%
%The graphs , known as phylogenetic trees, denote the ancestral relationships between the individuals, thus providing a window into the evolution of an organism.
%
%Recent work has also revealed discrepancies between phylogenies of individual, independently-evolving genes and species trees
%
%I develop a simple model to describe a cophylogenetic process

Organismic symbioses, or interactions between individuals of two or more different species, constitute a fundamental aspect of many biological systems and occur across the domains of life. Symbiotic relationships differ on multiple spectrums, such as degrees of cooperation (parasitism vs. mutualism), fidelity (generalists utilizing multiple symbionts vs. specialized, species--species-specific associations), and obligation (optional vs. vital to survival). Understanding how these relationships evolve and, in particular, how symbionts affect their partners' evolution---that is, the coevolutionary processes driven by symbiotic interactions---remain important questions in evolutionary biology.

Phylogenies, or evolutionary trees, can provide a valuable perspective to such evolutionary processes. 

\textcite{Felsenstein:1981} introduced a likelihood

Bayesian inference relies on the principle of Bayes' theorem \parencite{Bayes:1763}; specifically, that the posterior probability of a model's parameters given some data $P\left(\theta|D\right) \propto P\left(D|\theta\right) P\left(\theta\right)$ for a constant dataset, where $\theta$ is the model parameters and $D$ the data. The likelihood, $P\left(D|\theta\right)$, is the probability of simulating the observed data under the model parameters, and the prior, $P\left(\theta\right)$, is the probability of the model parameters. 

An important observation is that host--symbiont phylogenies rarely mirror each other perfectly, even in 

\begin{figure}
\centering
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0.5)(0.5,0.5)
\psline[linecolor=blue](0.5,0.3)(1,0.3)
\psline[linecolor=blue](0.5,0.7)(1,0.7)
\psline[linecolor=blue](0.5,0.3)(0.5,0.7)
\psline[linecolor=red](0.4,0.4)(1,0.4)
\psline[linecolor=red](0.4,0.8)(1,0.8)
\psline[linecolor=red](0.4,0.4)(0.4,0.8)
\psline[linecolor=red,arrows=-o](0,0.6)(0.4,0.6)
\end{pspicture}
\caption{cospeciation}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=red](0,0.1)(1,0.1)
\psline[linecolor=red,arrows=*-](0.5,0.1)(0.5,0.3)
\psline[linecolor=red](0.5,0.3)(1,0.3)
\end{pspicture}
\caption{duplication}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=blue](0,0.4)(1,0.4)
\psline[linecolor=red](0,0.1)(1,0.1)
\psline[linecolor=red](0.5,0.1)(0.5,0.5)
\psline[linecolor=red,arrows=->,arrowsize=0.1](0.5,0.1)(0.5,0.35)
\psline[linecolor=red](0.5,0.5)(1,0.5)
\end{pspicture}
\caption{host-shift}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=red](0,0.1)(0.8,0.1)
\rput(0.8,0.1){\large\textcolor{red}{\textsf{x}}}
\end{pspicture}
\caption{loss}
\end{subfigure}
\caption{Diagrams of the four cophylogenetic events considered in my model, where the host organism's phylogeny is in blue and the symbiont phylogeny in red. All events are branch events (i.e., they can occur at any point along a branch) except cospeciation, which is a nodal event (and thus can occur only when the host speciates).}
\end{figure}

\section*{Materials and Methods}

%\begin{equation}
%P\left(M,\Theta_M|D\right) \propto \int\limits_{T_H \times T_S} \overbrace{P\left(d_H|t_H,\theta_H\right)}^{\mathclap{\text{Felsenstein likelihood for host tree}}} \underbrace{P\left(d_S|t_S,\theta_S\right)}_{\mathclap{\text{Felsenstein likelihood for symbiont tree}}} P\left(d_M|M,\theta_M,t_H,t_S\right) P\left(M\right) \textrm{d}T_H \times T_S 
%\end{equation}

%\begin{equation}
%P\left(m,\theta|D\right) \propto \int\limits_{T_\textrm{H} \times T_\textrm{S}}  P\left(d_\textrm{M}|m, \theta\right) P\left(t_\textrm{H}|d_\textrm{H}\right) P\left(t_\textrm{S}|d_\textrm{S}\right) P\left(\theta\right) \textrm{d} t_\textrm{H},t_\textrm{S} 
%\end{equation}
%
%This integral can be approximated via a \ac{MCMC} analysis.
%
%Nevertheless, it is possible to model these events on the principle of observation; i.e., what specific event or events are we observing.
%
%For example, we can consider molecular evolution.

Given data $D = \left(d_H,d_S,d_R\right)$, the host sequence data, the symbiont sequence data, and the leaf--leaf associations between the host and symbiont trees, respectively, I define the posterior probability of the host tree $H$ and the symbiont tree $S$, both with branch lengths, and the reconstruction $R$ mapping internal nodes of $S$ to $H$ as
\begin{equation}
P\left(H,S,R|D\right) \propto \int P\left(d_S\right|S) P\left(S|H,R,d_R,\theta\right) P\left(H|d_H\right) P\left(R\right) P\left(\theta\right) \, \dif \theta
\end{equation}
where parameters $\theta = \left(\lambda,\tau,\mu\right)$, the duplication, host-switch, and loss rates, respectively. The probability of a tree $T$ given sequence data $D$ is $P\left(T|D\right) \propto P\left(D|T\right) P\left(T\right)$, where the likelihood is generally calculated with \citeapos{Felsenstein:1981} algorithm or a more complex multi-locus model, e.g. the multispecies coalescent \parencite{Heled:2010a}, and the prior with either a coalescent model (e.g., constant size) or a birth-death speciation model \parencite{Gernhard:2008}. In this case, the prior on the symbiont tree is represented instead by the reconstruction likelihood $P\left(S|H,R,d_R,\theta\right)$. Note that I treat the reconstruction parameters $\theta$ as nuisance variables and integrate them out when calculating the posterior probability of the reconstruction.

\subsection*{Estimation of the Reconstruction Likelihood}

I implemented the described model in the Java language as a plugin for the BEAST program \parencite{Drummond:2012} for Bayesian evolutionary analysis via \ac{MCMC} simulation. There were several advantages , including  the existing \ac{MCMC} framework and evolutionary library, the number of models, and the ability to utilize any future models developed for BEAST.

The source code is available under the GNU General Public License version 3 at \url{http://www.github.com/phylocomputing/BECKY}.

\section*{Results}

\section*{Discussion}

\section*{Conclusions and Future Work}

%The effects of the cophylogenetic likelihood on mixing of the MCMC chain needs to be seriously investigated.

There are several avenues upon to expand the simple cophylogeny model. 

\printbibliography

\end{document}
