% !TEX TS-program = pdflatex
% !BIB TS-program = biber
\documentclass[12pt,letterpaper]{article}

% Layout and formatting
\usepackage[margin=1in]{geometry}
\usepackage{acronym}
\usepackage{csquotes}
\usepackage{fixltx2e}
\usepackage{url}
\usepackage{mathptmx}
\usepackage{setspace}
\usepackage{units}
\frenchspacing

% Graphics
\usepackage{graphicx}
\usepackage[pdf]{pstricks}
\psset{unit=1in,linewidth=0.02,arrows=C-C,arrowsize=0.2}

% Bibliography
\usepackage[american]{babel}
\usepackage[backend=biber,style=apa]{biblatex}
\DeclareLanguageMapping{american}{american-apa}
\bibliography{bibliography.bib}
\newcommand{\citeapos}[2]{\citeauthor{#1}'s #2 (\citeyear{#1})}
\usepackage{doi}

% Figures, tables, and captions
\usepackage{float}
\usepackage{booktabs}
\usepackage[hypcap]{caption}
\captionsetup{labelfont=bf,font=small,labelsep=period}
\usepackage{subcaption}
\usepackage[rightcaption]{sidecap}
\usepackage[plain]{fancyref}

% Math
\usepackage{amsmath}
\usepackage[algoruled]{algorithm2e}
\usepackage{mathtools}
\usepackage{commath}

\title{Bayesian Reconstruction of Coevolutionary Histories}

% Acronyms
\acrodef{GMTC}{geographic mosaic theory of coevolution}
\acrodef{GTR}{general time reversible}
\acrodef{MCMC}{Markov chain Monte Carlo}

% Reusable figures
\newcommand{\pscophylogeny}{
\begin{pspicture}(18,12)
\psset{unit=0.5cm,linewidth=0.2}
\psline[linecolor=blue](0,0)(10,10)
\psline[linecolor=blue](4,0)(2,2)
\psline[linecolor=blue](8,0)(4,4)
\psline[linecolor=blue](12,0)(14,2)
\psline[linecolor=blue](16,0)(8,8)
\psline[linecolor=red](1,0)(11,10)
\psline[linecolor=red,arrows=-o](17,0)(9,8)
\psline[linecolor=red,arrows=-o](13,0)(15,2)
\psline[linecolor=red](7,3)(14,3)
\psline[linecolor=red,arrows=<-](10,3)(14,3)
\psline[linecolor=red](10,0)(7,3)
\psline[linecolor=red,arrows=-o](9,0)(5,4)
\psline[linecolor=red,arrows=-o](4,1)(3,2)
\rput{135}(4,1){\LARGE\textcolor{red}{\textsf{\textbf{x}}}}
\psline[linecolor=red](18,0)(17,1)
\psline[linecolor=red,arrows=*-](16,1)(17,1)
\end{pspicture}
}

\begin{document}

\begin{titlepage}
\null
\vfil
\let\newpage\relax
\maketitle
\vfil
\centering
\pscophylogeny
\vfil
\thispagestyle{empty}
\end{titlepage}

\newpage

\doublespacing

\section*{Introduction}

Organismic symbioses, or interactions between individuals of two or more different species, constitute a fundamental aspect of many living systems and occur across the biological spectrum. The characteristics of symbiotic relationships can vary greatly, particularly in degrees of cooperation (parasitism vs. mutualism), fidelity (generalized symbionts utilizing multiple species vs. specialized, species-specific symbionts), and obligation (optional vs. vital to survival) \parencites{Charleston:2002}{Oberprieler:2004}{Becerra:2007}{Beinart:2012}{HoyalCuthill:2012}{Thompson:2012}{Faria:2013}. Understanding how these relationships evolve and, in particular, how symbionts affect their partners' evolution---that is, the coevolutionary processes driven by symbiotic interactions---remain important questions in ecology and evolutionary biology. 

Phylogenies, or evolutionary trees, can provide a valuable perspective to evolutionary processes. The topology of the tree depicts the set of speciation events that gave rise to those taxa and thus their ancestral relationships \parencite{Baum:2008}. Often, the lengths of branches indicate the number of substitutions (e.g., genetic mutations) accumulated or the amount of time passed at that particular lineage \parencite{Baum:2008}. By mapping traits such as physical attributes or geographic locations onto a known phylogeny, we can use the tree to learn about the mechanisms under which these traits change \parencites{Lemey:2009}{Lemey:2010}{Segraves:2010}.

Molecular sequences are the primary source of data for constructing phylogenies \parencite{Baum:2008}. With the advancements of DNA sequencing technology over the last few decades, considerable effort has been invested in developing robust statistical methods for inferring phylogenies from sequence alignments (of both nucleotides and amino acids) \parencite{Felsenstein:2005}. An early technique involved optimizing the tree topology under the parsimony criterion \parencite{Fitch:1971}. The most parsimonious, or optimal, tree is the \enquote{simplest} explanation for a sequence alignment---the tree involving the least number of substitutions. Following his criticism of parsimony methods \parencite{Felsenstein:1978}, \textcite{Felsenstein:1981} introduced an algorithm calculate the likelihood of a tree given a substitution model for evolution and sequence data. Similar to maximum parsimony, maximum likelihood techniques optimize a tree under Felsenstein's likelihood function, often simultaneously estimating parameter values for the substitution model \parencite{Felsenstein:2005}. The substitution model enables sophisticated models of evolution to be used to infer phylogenies, such as models aware of the underlying molecular evolutionary processes.

Many recent advances in phylogenetics have focused on methodologies utilizing Bayesian inference \parencites{Ronquist:2012}{Drummond:2012}. Bayesian inference takes its name from Bayes' theorem \parencite{Bayes:1763} and relies on the principle that the posterior probability of a model's parameters~$\theta$ given some data~$D$ is $P\left(\theta|D\right) \propto P\left(D|\theta\right) P\left(\theta\right)$ for a constant dataset. The likelihood, $P\left(D|\theta\right)$, is the probability of simulating the observed data under the model parameters (e.g., \citeauthor{Felsenstein:1981}'s \citeyear{Felsenstein:1981} algorithm), and the prior, $P\left(\theta\right)$, is the probability of the model parameters without considering the data. The prior can be a mechanism to apply previous knowledge to a new analysis; for example, constructing a prior distribution on the mutation rate representative of the rates reported in previous literature will \enquote{bias} the posterior to favor those rates. When explicit prior knowledge is not available, an uninformative prior may be applied.

Often we are interested in only some of the parameters involved in an analysis (e.g., just the tree topology), in which case we can calculate the posterior probability of specific values for those parameters by integrating over all possible values of the remaining \enquote{nuisance} parameters (branch lengths, mutation rate, etc.). Due to the complexity of the models, the resulting multi-dimensional integral cannot be evaluated directly but instead approximated via \ac{MCMC}. The Metropolis--Hastings algorithm \parencites{Metropolis:1953}{Hastings:1970} is a common implementation of the \ac{MCMC} method in phylogenetic inference

Although there have been criticisms of Bayesian \ac{MCMC} methodologies \parencites{Felsenstein:2005}{Kolaczkowski:2009}, they remain popular for \parencite{}.

\textcite{Haffner:1988} were the first to apply phylogenetic methods  

An important observation is that host--symbiont phylogenies rarely mirror each other perfectly, even in 

There are a number of , e.g., Tarzan \parencite{Merkle:2005}, CoRe-PA \parencite{Merkle:2010}, Jane \parencite{Conow:2010}, TreeMap \parencite{Charleston:2011}, and AnGST \parencite{David:2011}.

Brooks Parsimony Analysis \parencite{Brooks:1981} and the coalescent \parencite{Rannala:2003} , 

A Bayesian \ac{MCMC} method for inferring cophylogenies also lends a number of advantages.

A primary objective of parsimony approaches to the cophylogeny problem is developing heuristics to find optimal reconstructions quickly. For example, \textcite{Charleston:1998} proposed the \enquote{jungle} concept and the program Jane utilizes a genetic algorithm to find optimal timings \parencite{Conow:2010}. This problem becomes much less of a concern in an \ac{MCMC} framework, because the parameter space is being explored as opposed to maximized (although it is still critical to employ effective operators for efficient mixing of the chain). However, it is important to note that \ac{MCMC} methods carry a much greater computational expense. 

I describe a simple cophylogeny model that approximates and implement it in a Bayesian framework. 

\begin{figure}
\centering
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0.5)(0.5,0.5)
\psline[linecolor=blue](0.5,0.3)(1,0.3)
\psline[linecolor=blue](0.5,0.7)(1,0.7)
\psline[linecolor=blue](0.5,0.3)(0.5,0.7)
\psline[linecolor=red](0.4,0.4)(1,0.4)
\psline[linecolor=red](0.4,0.8)(1,0.8)
\psline[linecolor=red](0.4,0.4)(0.4,0.8)
\psline[linecolor=red,arrows=-o](0,0.6)(0.4,0.6)
\end{pspicture}
\caption{cospeciation}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=red](0,0.1)(1,0.1)
\psline[linecolor=red,arrows=*-](0.5,0.1)(0.5,0.3)
\psline[linecolor=red](0.5,0.3)(1,0.3)
\end{pspicture}
\caption{duplication}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=blue](0,0.4)(1,0.4)
\psline[linecolor=red](0,0.1)(1,0.1)
\psline[linecolor=red](0.5,0.1)(0.5,0.5)
\psline[linecolor=red,arrows=->,arrowsize=0.1](0.5,0.1)(0.5,0.35)
\psline[linecolor=red](0.5,0.5)(1,0.5)
\end{pspicture}
\caption{host-switch}
\end{subfigure}
\begin{subfigure}[b]{0.2\textwidth}
\centering
\begin{pspicture}(1,1)
\psline[linecolor=blue](0,0)(1,0)
\psline[linecolor=red](0,0.1)(0.8,0.1)
\rput(0.8,0.1){\large\textcolor{red}{\textsf{x}}}
\end{pspicture}
\caption{loss}
\end{subfigure}
\caption{Diagrams of the four cophylogenetic events considered in my model, where the host organism's phylogeny is in blue and the symbiont phylogeny in red. All events are branch events (i.e., they can occur at any point along a branch) except cospeciation, which is a nodal event (and thus can occur only when the host speciates).}
\label{fig:events}
\end{figure}

\begin{SCfigure}
\centering
\pscophylogeny
\caption{An example of a coevolutionary history, with the host phylogeny in blue, the symbiont phylogeny in red, and symbols corresponding with the four events depicted in \Fref{fig:events}.}
\label{fig:cophylogeny}
\end{SCfigure}

Finally, I apply 

\section*{Materials and Methods}

%\begin{equation}
%P\left(M,\Theta_M|D\right) \propto \int\limits_{T_H \times T_S} \overbrace{P\left(d_H|t_H,\theta_H\right)}^{\mathclap{\text{Felsenstein likelihood for host tree}}} \underbrace{P\left(d_S|t_S,\theta_S\right)}_{\mathclap{\text{Felsenstein likelihood for symbiont tree}}} P\left(d_M|M,\theta_M,t_H,t_S\right) P\left(M\right) \textrm{d}T_H \times T_S 
%\end{equation}

%\begin{equation}
%P\left(m,\theta|D\right) \propto \int\limits_{T_\textrm{H} \times T_\textrm{S}}  P\left(d_\textrm{M}|m, \theta\right) P\left(t_\textrm{H}|d_\textrm{H}\right) P\left(t_\textrm{S}|d_\textrm{S}\right) P\left(\theta\right) \textrm{d} t_\textrm{H},t_\textrm{S} 
%\end{equation}
%
%This integral can be approximated via a \ac{MCMC} analysis.
%
%Nevertheless, it is possible to model these events on the principle of observation; i.e., what specific event or events are we observing.
%
%For example, we can consider molecular evolution.

Given data~$D = \left(d_H,d_S,d_R\right)$, the host sequence data, the symbiont sequence data, and the leaf--leaf associations between the host and symbiont trees, respectively, I define the posterior probability of the host tree~$H$ and the symbiont tree~$S$, both with branch lengths, and the reconstruction~$R$ mapping internal nodes of $S$ to $H$ as
\begin{equation}
P\left(H,S,R|D\right) \propto \int P\left(d_S\right|S) P\left(S|H,R,d_R,\theta\right) P\left(H|d_H\right) P\left(R\right) P\left(\theta\right) \, \dif \theta
\label{eq:cophylogenyposterior}
\end{equation}
where parameters $\theta = \left(\lambda,\tau,\mu\right)$, the duplication, host-switch, and loss rates, respectively, and are integrated out. The probability of a tree~$T$ given sequence data~$D$ is $P\left(T|D\right) \propto P\left(D|T\right) P\left(T\right)$, where the likelihood is generally calculated with \citeapos{Felsenstein:1981} algorithm or a more complex multi-locus model, e.g. the multispecies coalescent \parencite{Heled:2010a}, and the prior with either a coalescent model \parencite{Kingman:1982} or a birth-death speciation model \parencite{Gernhard:2008}. In this case, the prior on the symbiont tree is represented instead by the reconstruction likelihood $P\left(S|H,R,d_R,\theta\right)$.

\begin{algorithm}
\caption{Key algorithm.}
\KwIn{Host Tree $H$, Symbiont Tree $S$, }
\end{algorithm}

\begin{SCfigure}[1.5]
\centering
\begin{pspicture}(18,12)
\psset{unit=0.25cm,linewidth=0.3}
\psline(0,0)(10,10)
\psline(4,0)(2,2)
\psline(8,0)(4,4)
\psline(12,0)(14,2)
\psline(16,0)(8,8)
\rput(0,-0.75){\small\textbf{1}}
\rput(4,-0.75){\small\textbf{2}}
\rput(8,-0.75){\small\textbf{3}}
\rput(12,-0.75){\small\textbf{4}}
\rput(16,-0.75){\small\textbf{5}}
\rput(16,-0.75){\small\textbf{5}}
\rput(1.5,2.5){\small\textbf{6}}
\rput(3.5,4.5){\small\textbf{7}}
\rput(14.5,2.5){\small\textbf{8}}
\rput(7.5,8.5){\small\textbf{9}}
\end{pspicture}
\caption{In the depicted topology, node 6 is a descendant of nodes 7 and 9, an ancestor of nodes 1 and 2, and a cousin of all other nodes.}
\label{fig:nodalrelationships}
\end{SCfigure}

\paragraph*{Calculating the event probabilities.} 

I model the three 

: we may be observing a loss event on that particular branch, or, alternatively, no event at this branch and some series of loss events along its child branches. Hence, the probability of observing a loss event along a branch $b(N)$ of $H$ is a summation over all permutations of loss events that . 

\paragraph*{Implementation in a Bayesian framework.}

I implemented the described model in the Java language as a plugin for the BEAST program \parencite{Drummond:2012} for Bayesian evolutionary analysis via \ac{MCMC} analysis. There were several advantages to integrating my work into BEAST, particularly the existing \ac{MCMC} framework and evolutionary library consisting of numerous models.

To provide support for rate variation over the symbiont phylogeny, a clock model provides the rate. Due to the additional free parameter, I fix the loss rate~$\mu$ at $1$, such that the duplication and host-switch rates $\lambda$ and $\tau$ are normalized against it.

The two priors established in \Fref{eq:cophylogenyposterior} remain to be defined. I assume a uniform distribution over $R$, so $P(R) \propto 1$.

Finally, I define a simple \textsc{host shift operator} to . An internal node $n_S$ from $S$ is selected at random and is assigned a new host node from the set of host lineages existing at time $t = height(n_S)$.\

The source code is available under the GNU General Public License at \url{http://www.github.com/phylocomputing/BECKY}. I provide a Python script to help modify BEAST XML files to i.

\paragraph*{Evaluation of simulated data.}

I simulate the host tree $H$ under the constant-size coalescent \parencite{Kingman:1982} as implemented in BEAST \parencite{Drummond:2012}

Two simulations

%\paragraph*{Case study: cycad--weevil coevolution}



\section*{Results}

\paragraph*{Simulated data.}

%\paragraph*{Cycad--weevil coevolution.}

\section*{Discussion}

A important question from the biological perspective is, how fine a scale ? 

In research on gopher--louse

One of the inherent challenges with the cycad--weevil dataset was . 

Some cophylogeny reconstruction programs, such as Jane \parencite{Conow:2010}, consider the \enquote{} event

The greater takeaway, however, is that to to perform a complete, cophylogenetic analysis on a datset

\section*{Conclusions and Future Work}

An immediate problem is the inability for the and will necessitate . Along these lines, the effects of the cophylogeny likelihood on mixing of the \ac{MCMC} calls for serious investigation. 

There are several avenues upon to expand the simple cophylogeny model. 

\textcite{Charleston:2002} and \textcite{Faria:2013} both find preferential host-switching to be a phenomenon . 

I discussed . \textcite{Yang:2010} developed a Bayesian \ac{MCMC} methodology to determine the species delimitations between a set of individuals. Maintaining 

The model could also be expanded to support geographical data. In BEAST there are implementations of both discrete and continuous models for phylogeography \parencites{Lemey:2009}{Lemey:2010}, so simply restricting the location of an ancestral symbiont to be that of its host.

\ac{GMTC} . Ultimately, it may be possible , and thus to test the \ac{GMTC} against coevolutionary histories.

\printbibliography

\end{document}
